src/phylo/phylogeny.jl

```diff
@@ -0,0 +1,230 @@
+# phylo/phylogeny.jl
+# ==================
+#
+# Types and methods for phylogenetic trees.
+#
+# Part of the Bio.Phylo module.
+#
+# This file is a part of BioJulia. License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+
+
+# The Phylogeny Type
+# -------------------
+
+"""
+Phylogeny represents a phylogenetic tree.
+
+Fields:
+
+    - graph: A LightGraphs directional graph.
+
+    - ntaxa: A field to keep track of the number of tips in the phylogeny.
+
+    - rooted: A boolean field to keep track of whether the phylogeny is considered
+    rooted or not.
+
+    - rerootable: A boolean field to keep track of whether the phylogeny may be
+    rerooted or not.
+"""
```

Comment:
I think docstrings do not need to mention the internal fields. These docstrings may send a wrong signal that these fields are accessible to the user. If you intended to use them internally, comments rather docstrings are suitable for developers.